app.js

```diff
@@ -192,34 +247,116 @@ async function sendEmail(courseMap, recipients) {
+        return
+      }
+  
+      const courseDetailsHTML = formatCourseInfoHTML(courseCode, courseMap[courseCode])
   
-  let courseDetailsHTML = ''
+      const response = await transporter.sendMail({
+        from: EMAIL_ADDR,
+        to: EMAIL_ADDR,
+        bcc: emails,
+        subject: `Your course selection info for ${courseCode}`,
+        html: fs.readFileSync(__dirname + '/templates/coursesAvailableEmail.html', 'utf8').replace('${INSERT_INFO_HERE}', courseDetailsHTML),
+      })
 
-  Object.entries(courseMap).forEach(([courseCode, details]) => {
-    courseDetailsHTML += formatCourseInfoHTML(courseCode, details)
+      // Filters out our email address from the response
+      response.accepted = response.accepted.filter(email => email != EMAIL_ADDR)
+      response.rejected = response.rejected.filter(email => email != EMAIL_ADDR)
+  
+      console.warn(`batch ${i + 1} - failed emails`, response.rejected)
+      console.info(`batch ${i + 1} - successful emails`, response.accepted)
+  
+      response.accepted.forEach((email) => {  
+        ddb.update({
+          TableName: 'Courses',
+          Key: { 'CourseCode': courseCode },
+          UpdateExpression: 'ADD #email :sentTimestamp',
+          ExpressionAttributeNames: {
+            '#email': email
+          },
+          ExpressionAttributeValues: {
+            ':sentTimestamp': ddb.createSet([new Date().setHours(0, 0, 0, 0)]),
+          }
+        }).promise().catch ((err) => {
+          console.error(`Error updating ${email} timestamp in dynamodb`, err)
+        })
+      })
+    })())
   })
 
-  if (!courseDetailsHTML) {
-    console.warn('No courses were available so no email was sent')
-    return
-  }
+  await Promise.all(emailPromises)
+}
 
-  const info = await transporter.sendMail({
-    from: EMAIL_ADDR, 
-    to: recipients, 
-    subject: 'Your course selection info', 
-    html: fs.readFileSync(__dirname + '/templates/coursesAvailableEmail.html', 'utf8').replace('${INSERT_INFO_HERE}', courseDetailsHTML),
-  });
+function formatCourseInfoHTML(courseCode, courseInfo) {
+  if (!courseInfo.length) {
+    return ''
+  }
 
-  console.info('emails sent to', info.accepted)
-  console.info('emails did not send to', info.rejected)
+  return `<p><b>${courseCode}</b> has sections available</p> <p>${courseInfo}</p> <br>`
 }
 
-function formatCourseInfoHTML(courseCode, courseInfo) {
-  if (courseInfo.length) {
-    return `<p><b>${courseCode}</b> has sections available</p> <p>${courseInfo}</p> <br>`
+/**
+ * Queries dynamo db to find the user's to notify for each course
+ * 
+ * @param {Array} courseCodes - Array of course codes of available courses
+ * @returns {Object} - Keys correspond to the course codes and the value 
+ * is an array of emails to send the course to
+ */
+async function getEmailsCourseCodesFromDyanmo(courseCodes) {
+  let coursesQuery = []
+
+  // Formats all the course codes for DynamoDB
+  courseCodes.forEach(cc => coursesQuery.push({
```

Comment:
get out of the habit of defining the array (using let) and then pushing. Defining and mapping is more efficient

```suggestion
  const coursesQuery = courseCodes.map(cc => 'CourseCode': cc)
```

Reply:
smart!